### PR TITLE
feat: bootstrap token limit

### PR DESCRIPTION
**Pull request by sergelogvinov**
Token can use only for bootstrap nodes

split PR #3807

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

**Comment by smira** [comment]
> Token can use only for bootstrap nodes

does it mean that it limits bootstrap token use to the nodes which are being bootstrapped right now? the ones which haven't bootstrapped yet? what would it do if the node was wiped and the `Node` resource is already present?

**Comment by sergelogvinov** [comment]
All will work as before.

I've changed the bootstrap token group system:bootstrappers -> system:bootstrappers:nodes
You can use bootstrap token not only for join nodes. Default group system:bootstrappers can affect another solutions base on bootstrap tokens.

File: internal/app/machined/pkg/controllers/k8s/templates.go [hunk 1]
```diff
@@ -37,16 +37,19 @@ stringData:
   token-id: "{{ .Secrets.BootstrapTokenID }}"
   token-secret: "{{ .Secrets.BootstrapTokenSecret }}"
   usage-bootstrap-authentication: "true"
+
+  # Extra groups to authenticate the token as. Must start with "system:bootstrappers:"
+  auth-extra-groups: system:bootstrappers:bootstrap-node-token,system:bootstrappers:nodes
```
Comment (smira) [inline internal/app/machined/pkg/controllers/k8s/templates.go:42]:
```suggestion
  auth-extra-groups: system:bootstrappers:nodes
```

**Review by smira — DISMISSED** [review]

**Review by smira — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**